Scripts/Python/ki/xKIChat.py

```diff
@@ -103,6 +100,10 @@ def __init__(self, StartFadeTimer, ResetFadeState, FadeCompletely, GetCensorLeve
         # Add the commands processor.
         self.commandsProcessor = CommandsProcessor(self)
 
+        # Set a fake player name to avoid errors.
+        self.playerName = None
+        self._chatMentionRegex = None
```

Comment:
```suggestion
```

Overrides the "default" (read: error case) regex, complicating the code below. The player name is guaranteed to be set by `OnAccountUpdate()` before any chat can come in because there is no connection to any game (read: chat) server when the player is unset.

Reply:
I changed the playerName property setter not to set the _chatMentionRegex if the value was None (it seemed a little silly to always have to compile a "fake" regex the first time when we will never want it or use it), and I added the `self._chatMentionRegex = None` because I like variable declarations... even when not strictly necessary. I can get rid of the declaration, though, and simplify the mention check cases you noted if we are certain AddChatLine will never be called without a real playerName set. Is it important to do the default/error/fake playerName set process, though?

Reply:
I wanted it to be that way for consistency... when you return to the character selection screen, `OnAccountUpdate()` will be called again, and `PtGetClientName()` will return an empty string. So, it's important in my mind to handle "crap, there is an empty string here" and do it the same way each time. Minimal surprises and all. Once you login to another avatar, the same thing happens, except `PtGetClientName()` will return the avatar name. Related: the engine itself will specify that KI messages from unknown players are from "Anonymous Coward", which is where I got that.

Reply:
Hm, alright. I'll remove the if check in the setter. I guess the extra computation is minimal.